pkg/controller/clone/planner.go

```diff
@@ -414,6 +418,14 @@ func (p *Planner) computeStrategyForSourceSnapshot(ctx context.Context, args *Ch
 		p.fallbackToHostAssisted(args.TargetClaim, res, NoVolumeExpansion, MessageNoVolumeExpansion)
 		return res, nil
 	}
+
+	// Lastly, do volume mode validation to determine whether to use dumb or smart cloning
+	if !SameVolumeMode(vsc.Spec.SourceVolumeMode, args.TargetClaim) {
+		p.fallbackToHostAssisted(args.TargetClaim, res, IncompatibleVolumeModes, MessageIncompatibleVolumeModes)
+		args.Log.V(3).Info("Volume modes not compatible for advanced clone - Snapshot")
```

Comment:
Maybe message should be similar to the provisioners print? something like:
"source and target volume modes differs, need to fall back to host assisted"